commit f81e5b71ce78f33250347914dacc75c8463bf102
Author: Breno Leitao <breno.leitao@gmail.com>
Date:   Wed Mar 29 15:22:38 2017 -0300

    include: Check for previous declaration of uintptr_t
    
    Adding a extra check before declaring uintptr_t. Currently musl uses
    macro __DEFINED_uintptr_t once it defines uintptr_t type. Checking
    this macro before defining it, and, defining it when uintptr_t is
    defined.
    
    Signed-off-by: Breno Leitao <breno.leitao@gmail.com>

--- a/third_party/cmocka/cmocka.h
+++ b/third_party/cmocka/cmocka.h
@@ -111,7 +111,7 @@ typedef uintmax_t LargestIntegralType;
     ((LargestIntegralType)(value))
 
 /* Smallest integral type capable of holding a pointer. */
-#if !defined(_UINTPTR_T) && !defined(_UINTPTR_T_DEFINED)
+#if !defined(_UINTPTR_T) && !defined(_UINTPTR_T_DEFINED) && !defined(__DEFINED_uintptr_t)
 # if defined(_WIN32)
     /* WIN32 is an ILP32 platform */
     typedef unsigned int uintptr_t;
@@ -137,6 +137,8 @@ typedef uintmax_t LargestIntegralType;
 
 # define _UINTPTR_T
 # define _UINTPTR_T_DEFINED
+# define __DEFINED_uintptr_t
+
 #endif /* !defined(_UINTPTR_T) || !defined(_UINTPTR_T_DEFINED) */
 
 /* Perform an unsigned cast to uintptr_t. */
